### PR TITLE
Remove bodies of D functions that are trivial wrappers of C functions

### PR DESCRIPTION
**Pull request by n8sh**
This also allows removal of some private function declarations.

### CODE REVIEW

**Comment by n8sh** [comment]
https://github.com/dlang/druntime/blob/9f0f6e49fb379841c7934b6049f87dab0adfe53c/src/core/memory.d#L409-L419
I don't see why the overload of `core.memory.GC.getAttr` that casts away const is impure but for now I'm maintaining the distinction. Ditto for several other similar function pairs in `core.memory.GC`.

**Comment by n8sh** [comment]
https://github.com/dlang/druntime/blob/9f0f6e49fb379841c7934b6049f87dab0adfe53c/src/core/runtime.d#L692-L694
How has this been compiling successfully until now? I thought being able to selectively import private members of other modules had been removed a long time ago. I guess not!

**Comment by n8sh** [comment]
test/profile is segfaulting on 32-bit linux and FreeBSD (Windows hasn't run yet). I don't immediately see why that would happen. Will have to look into it later.

**Comment by rainers** [comment]
> How has this been compiling successfully until now?

It is declared `package`.

**Review by andralex — APPROVED** [review]

**Review by wilzbach — APPROVED** [review]

**Comment by thewilsonator** [comment]
Hmm, this still has failures on the autotester. @wilzbach any ideas?

**Comment by wilzbach** [comment]
Druntime test suite is terrible, but that's the error:

```
diff both.def.exp ./generated/linux/debug/32/both.def || diff bothnew.def.exp ./generated/linux/debug/32/both.def
4,5c4,5
< 	_D4both3fooFkZPS4both3Num
< 	_D4both3Num6__ctorMFNckZS4both3Num
---
> 	_D4both3fooFkZPSQo3Num
> 	_D4both3Num6__ctorMFNckZSQxQu
diff bothgc.log.exp ./generated/linux/debug/32/bothgc.log
make[2]: Leaving directory '/home/braddr/sandbox/at-client/pull-4207884-Linux_64_32/druntime/test/profile'
posix.mak:338: recipe for target 'test/profile/.run' failed
make[1]: *** [test/profile/.run] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by thewilsonator** [comment]
Removing automerge.

File: src/object.d [hunk 1]
```diff
@@ -318,12 +318,7 @@ bool opEquals(const Object lhs, const Object rhs)
     assert(obj1 != obj2);
 }
 
-private extern(C) void _d_setSameMutex(shared Object ownee, shared Object owner) nothrow;
-
-void setSameMutex(shared Object ownee, shared Object owner)
-{
-    _d_setSameMutex(ownee, owner);
-}
+extern(C) pragma(mangle, "_d_setSameMutex") void setSameMutex(shared Object ownee, shared Object owner);
```
Comment (n8sh) [inline src/object.d:321]: I'm going to have to walk this one back for now to avoid breaking some tests.

File: src/core/memory.d [hunk 2]
```diff
@@ -786,35 +688,23 @@ struct GC
...
 
     /**
      * Returns runtime stats for currently active GC implementation
      * See `core.memory.GC.Stats` for list of available metrics.
      */
-    static Stats stats() nothrow
-    {
-        return gc_stats();
-    }
+    pragma(mangle, "gc_stats") static Stats stats() nothrow;
```
Comment (n8sh) [inline src/core/memory.d:701]: Remarkably/worryingly this change by itself is enough to cause the linux32 build to consistently segfault in the profilegc test. Backing out this and a few other changes in `core.memory` will probably cause the test suite to pass but I'd like to understand why this is happening.

Comment (thewilsonator) [reply src/core/memory.d]: issue number?

**Pull request merged**